source/common/router/upstream_request.cc

```diff
@@ -94,6 +94,20 @@ UpstreamRequest::~UpstreamRequest() {
         FilterUtility::percentageOfTimeout(response_time, parent_.timeout().per_try_timeout_));
   }
 
+  // Ditto for request/response size histograms.
+  Upstream::ClusterRequestResponseSizeStatsOptRef req_resp_stats_opt =
+      parent_.cluster()->requestResponseSizeStats();
+  if (req_resp_stats_opt.has_value()) {
+    auto& req_resp_stats = req_resp_stats_opt->get();
+    req_resp_stats.upstream_rq_headers_size_.recordValue(parent_.downstreamHeaders()->byteSize());
```

Comment:
As long as we're improving these stats, should 100 continue headers be included in header size?
strange that no one cares about trailers, but I guess we can add them if anyone wants them, I'd think we'd want to keep them distinct from headers in any case.

Reply:
We are including the 100 continue headers right? In decodeHeaders() we do:

```
  response_headers_size_ = headers->byteSize();
  ```
  
  unconditionally. 
  
  I am not crazy about trailers, so we can add them later maybe?

Reply:
I take back my comment – we are not accounting in `decode100ContinueHeaders()`. I think we should... Adding it.

Reply:
My comment about punting on accounting for trailers remains though.